### PR TITLE
Do not delete master frames on ads when there is no ad content to display

### PR DESCRIPTION
**Pull request by sriramkrish85**
https://github.com/ampproject/amphtml/issues/2414

### CODE REVIEW

File: builtins/amp-ad.js [hunk 1]
```diff
@@ -382,7 +382,11 @@ export function installAd(win) {
           }
           this.toggleFallback(true);
         }
-        this.element.removeChild(this.iframe_);
+        // Remove the iframe only if it is not the master.
+        if (!this.iframe_.name || this.iframe_.name.indexOf('_master') == -1) {
```
Comment (cramforce) [inline builtins/amp-ad.js]: There is definitely a name. No need to null check.

File: test/functional/test-amp-ad.js [hunk 2]
```diff
@@ -318,8 +318,65 @@ function tests(name, installer) {
...
+          placeholder.setAttribute('placeholder', '');
+          ad.appendChild(placeholder);
+          expect(placeholder.classList.contains('amp-hidden')).to.be.false;
+
+          const fallback = document.createElement('div');
+          fallback.setAttribute('fallback', '');
+          ad.appendChild(fallback);
+          return ad;
+        }).then(ad => {
+          sandbox.stub(
```
Comment (cramforce) [inline test/functional/test-amp-ad.js]: Set the name to something else

File: builtins/amp-ad.js (hunk 1 above)
Comment (sriramkrish85) [inline builtins/amp-ad.js]: Done

File: test/functional/test-amp-ad.js (hunk 2 above)
Comment (sriramkrish85) [inline test/functional/test-amp-ad.js]: Done.

**Comment by cramforce** [comment]
LGTM

**Pull request merged**